src/Forwarder.cpp

```diff
@@ -193,35 +172,40 @@ void Forwarder::forward_epics_to_kafka() {
     GenerateFakePVUpdateTimer->start();
   }
 
+  using namespace std::chrono;
+  std::atomic<MILLISECONDS> IterationExecutionDuration(0ms);
+  MetricsTimer MetricsTimerInstance(MILLISECONDS(200), main_opt,
+                                    IterationExecutionDuration,
+                                    KafkaInstanceSet);
+
+  MetricsTimerInstance.start();
+
   while (ForwardingRunFlag.load() == ForwardingRunState::RUN) {
-    auto do_stats = false;
-    auto t1 = CLK::now();
-    if (t1 - t_lf_last > MS(2000)) {
+    auto TimeAtStartOfLoop = STEADY_CLOCK::now();
```

Comment:
I feel that the algorithm for determining the next poll time can be simplified. See the following (pseudo) code:

```
auto NextPollTime = STEADY_CLOCK::now() + Dt;
while (true) {
  if (STEADY_CLOCK::now() > NextPollTime) {
    // Do polling
    NextPollTime = NextPollTime+ (int((STEADY_CLOCK::now() - NextPollTime) / Dt) + 1) * Dt;
  }
  std::this_thread::sleep_until(NextPollTime);
}
```

Reply:
It doesn't matter because the idea is to get rid of all of this logic and everything except the consumer poll from this loop when Dolica tackles the next ticket.